src/field/gen_fieldops.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+"""Code-generator for arithmetic operators on Field2Ds/Field3Ds
+
+This uses the jinja template in gen_fieldops.jinja to generate code
+for the arithmetic operators, and prints to stdout.
+
+The `Field` class provides some helper functions for determining how to
+pass a variable by reference or pointer, and how to name arguments in
+function signatures. This allows us to push some logic into the
+templates themselves.
+
+"""
+
+
+from __future__ import print_function
+
+try:
+    from builtins import object
+except ImportError:
+    pass
+
+from collections import OrderedDict
+from copy import deepcopy as copy
+import itertools
+
+import jinja2
```

Comment:
Is it worth putting this in a try/except block so that we can give some BOUT++ related instructions (e.g. you need to install jinja2 see this page in the manual for how)?